### PR TITLE
Get nova console logs from VMs and write them to a separate file

### PR DESCRIPTION
**Pull request by jframos**
#### Reviewers

@flopezag @pratid 
#### Description

This PR is about _CLAUDIA-5789_.
Nova Console-Log will be written only in logger file (DEBUG) before deleting VMs generated by each sanity test. This information will not be shown on Dashboard Test Details page, only in the logger file.
#### Testing

Locally (Spain2, Prague)

### CODE REVIEW

**Comment by flopezag** [comment]
LGTM

**Comment by pratid** [comment]
Why not moving the whole console log to a separate "console.{uuid}.log" and keep it until the next execution?

File: fiware-region-sanity-tests/README.rst [hunk 1]
```diff
@@ -308,6 +308,16 @@ additionally an HTML report ``test_results.html`` (or the same basename as
 the former) is generated from the given template (or the default found at
 ``resources/templates/`` folder).
 
+Apart from these, a log file will be written with all logged info in a
```
Comment (flopezag) [inline fiware-region-sanity-tests/README.rst]: I prefer change Apart from there by Additionally,

a log file is written

File: fiware-region-sanity-tests/README.rst [hunk 2]
```diff
@@ -308,6 +308,16 @@ additionally an HTML report ``test_results.html`` (or the same basename as
...
+Sanity Check execution, based on its handlers configuration
+(`resources/logging_sanitychecks.conf`). When test cases involve VM
+launching, just before deleting the VM, *FIHealth Sanity Checks*
+will try to get the Nova Console-Log of that VM and it will write
```
Comment (flopezag) [inline fiware-region-sanity-tests/README.rst]: tries to get... and it writes

File: fiware-region-sanity-tests/README.rst [hunk 3]
```diff
@@ -308,6 +308,16 @@ additionally an HTML report ``test_results.html`` (or the same basename as
...
+the content in a new file `novaconsole_{region_name}_{server_id}.log`.
+If the Console-Log is empty or it was impossible to be retrieved,
+the file will not be generated. These files are only generated whether
```
Comment (flopezag) [inline fiware-region-sanity-tests/README.rst]: the file is not generated. These files are only generated if the log level is set to _DEBUG_

File: fiware-region-sanity-tests/sanity_checks [hunk 4]
```diff
@@ -141,7 +141,12 @@ done
 }
 TESTS=${TESTS:-tests/regions}
 
-# Main
+# Delete all generated log files before each sanity check execution.
+echo "Deleting log files from previous executions"
+rm -f *.log
```
Comment (pratid) [inline fiware-region-sanity-tests/sanity_checks]: This pattern scares a little :-). Better `test_novaconsole_*.log`?

Comment (jframos) [inline fiware-region-sanity-tests/sanity_checks]: I am thinking about this. We have a problem if Sanity Checks are parallelized. We must delete only the logs of the given regions by params (where Sanity Checks will be executed).

Comment (flopezag) [inline fiware-region-sanity-tests/sanity_checks]: Yes, I just was thinking in the paralization when I see this line.

Comment (jframos) [inline fiware-region-sanity-tests/sanity_checks]: Removing only _'novaconsole'_ log files for the given regions in fd1fb6b.

**Comment by flopezag** [comment]
LGTM

**Comment by jframos** [comment]
DO NOT MERGE, it seems not to be working on Jenkins.

**Comment by jframos** [comment]
Fixed in ff37d91... Region name (in file name) must be lowercase.

File: fiware-region-sanity-tests/sanity_checks [hunk 5]
```diff
@@ -141,7 +144,14 @@ done
 }
 TESTS=${TESTS:-tests/regions}
 
-# Main
+# Delete all generated Nova Console-Log files for the given regions before each sanity check execution.
+echo "Deleting Nova Console-Log files from previous executions for the given regions (${REGION_NAME_LIST[@]})"
+for region in ${REGION_NAME_LIST[@]}; do
```
Comment (pratid) [inline fiware-region-sanity-tests/sanity_checks]: Although this is valid, please avoid using arrays in the shell script. Instead, a simple space-separated string value may suffice.

**Comment by pratid** [comment]
LGTM

**Pull request merged**